src/main/java/org/ozwillo/dcexporter/service/CkanService.java

```diff
@@ -37,33 +39,54 @@
         return licenses.stream().collect(Collectors.toMap(CkanLicense::getId, CkanLicense::getTitle));
     }
 
-    public CkanDataset getOrCreateDataset(String name, String title) {
+    public Map<String, String> getTags() {
+        CkanClient ckanClient = new CkanClient(ckanUrl);
+        List<CkanTag> tags = ckanClient.getTagList();
+        return tags.stream().collect(Collectors.toMap(CkanTag::getId, CkanTagBase::getName));
+    }
+
+    public CkanDataset getOrCreateDataset(DcModelMapping dcModelMapping) {
         CkanClient ckanClient = new CkanClient(ckanUrl, ckanApiKey);
 
         CkanDataset ckanDataset = null;
         try {
-            ckanDataset = ckanClient.getDataset(name);
+            ckanDataset = ckanClient.getDataset(dcModelMapping.getName());
         } catch (CkanException e) {
             // Not Found Error is an « expected » result
             // FIXME : this is a poor way to perform a search
             if (!e.getCkanResponse().getError().getType().equals("Not Found Error")) {
                 throw e;
             }
 
-            LOGGER.debug("Dataset {} already exists", name);
+            LOGGER.debug("Dataset {} already exists", dcModelMapping.getName());
         }
 
         if (ckanDataset == null) {
```

Comment:
Why don't you set all the available info (licence, ...) when it is a new CKAN dataset ?